server/src/main/java/org/elasticsearch/index/IndexSettings.java

```diff
@@ -246,10 +246,12 @@
     /**
      * Controls how many soft-deleted documents will be kept around before being merged away. Keeping more deleted
      * documents increases the chance of operation-based recoveries and allows querying a longer history of documents.
-     * If soft-deletes is enabled, an engine by default will retain all operations up to the global checkpoint.
+     * If soft-deletes is enabled, an engine by default will keep all soft-deleted documents in the most recent 4GB
+     * of Lucene index including live and soft-deleted documents. The extra required space depends on the deletion ratio.
      **/
-    public static final Setting<Long> INDEX_SOFT_DELETES_RETENTION_OPERATIONS_SETTING =
-        Setting.longSetting("index.soft_deletes.retention.operations", 0, 0, Property.IndexScope, Property.Dynamic);
+    public static final Setting<ByteSizeValue> INDEX_SOFT_DELETES_RETENTION_SIZE_SETTING =
+        Setting.byteSizeSetting("index.soft_deletes.retention.size", new ByteSizeValue(4, ByteSizeUnit.GB),
```

Comment:
this looks like a bit too high for a default. What about 512MB or 1GB? @bleskes any gut feelings?

Reply:
This is a good one. It's hard for me to come up with something good. Together with the time based policy we're evaluating it becomes easier (keep up to 24 hours but at most 5GB). Let's discuss this with the group?